src/WebViewShared.tsx

```diff
@@ -75,24 +85,29 @@ const createOnShouldStartLoadWithRequest = (
 
     /** Check if the url passes the origin whitelist */
     if (!_passesWhitelist(compiledWhiteList, url)) {
-      /** Check if the url passes the hardcoded deeplink blocklist */
-      const foundMatchInBlocklist = matchWithRegexList(defaultDeeplinkBlocklist, url)
-      if (!foundMatchInBlocklist) {
-        /** Check if the url passes the dynamic deeplink allow list */
-        const foundMatchInAllowlist = matchWithPrefixStringList(deepLinkWhitelist, url)
-  
-        if (foundMatchInAllowlist) {
-          Linking.canOpenURL(url).then((supported) => {
-            if (supported && isTopFrame) {
-              return Linking.openURL(url);
-            }
-            console.warn(`Can't open url: ${url}`);
-            return undefined;
-          }).catch(e => {
-            console.warn('Error opening URL: ', e);
-          });
-        } else {
-          console.warn(`Failed to pass default block list or whitelist deep link url: ${url}`);
+      const protocol = urlToProtocolScheme(url)
+      
+      /* Check that the protocol was properly parsed */
+      if (protocol !== null) {
```

Comment:
`protocol` can't be `null` here since invalid URL would throw in `_passesWhitelist`: https://github.com/ExodusMovement/react-native-webview/blob/3e4721819d602fd5730f65a7a5205469800edc76/src/WebViewShared.tsx#L51

Not a security issue. Don't mind leaving this in case the `_passesWhitelist` implementation changes in future versions.

Reply:
Or maybe we should wrap it in `try..catch` in `_passesWhitelist` right now. Would anything handle this exception?

Reply:
> Would anything handle this exception?

It would never reach that code due to the regex requiring the letter start.

https://github.com/ExodusMovement/react-native-webview/blob/48d7bf3bdd49b69aced7026f0f4bcb6798671cc1/src/WebViewShared.tsx#L23

So here it returns early:
https://github.com/ExodusMovement/react-native-webview/blob/48d7bf3bdd49b69aced7026f0f4bcb6798671cc1/src/WebViewShared.tsx#L49-L51

Reply:
Not bad 😅

Reply:
fixed in [f13e8b8](https://github.com/ExodusMovement/react-native-webview/pull/23/commits/f13e8b852829e8e31678de0d42bbd561b1deb717)